Amplify/Core/Configuration/AmplifyConfiguration.swift

```diff
@@ -40,20 +41,24 @@ public struct AmplifyConfiguration: Codable {
     /// Configurations for the Amplify Storage category
     let storage: StorageCategoryConfiguration?
 
+    let auth: AuthCategoryConfiguration?
+
     public init(analytics: AnalyticsCategoryConfiguration? = nil,
                 api: APICategoryConfiguration? = nil,
                 dataStore: DataStoreCategoryConfiguration? = nil,
                 hub: HubCategoryConfiguration? = nil,
                 logging: LoggingCategoryConfiguration? = nil,
                 predictions: PredictionsCategoryConfiguration? = nil,
-                storage: StorageCategoryConfiguration? = nil) {
+                storage: StorageCategoryConfiguration? = nil,
+                auth: AuthCategoryConfiguration? = nil) {
```

Comment:
Across this PR, there are various code constructs that amount to lists of the categories. Most (all?) of them appear to be lexicographically ordered by category name. To maintain this ordering, `Auth**` fields and parameters would come after `api`, and before `dataStore`. In this particular instance, the param would be come the third param to `init(...)`, not the last param.

Reply:
Updated